arachne/client/arachne/generators.py

```diff
@@ -0,0 +1,373 @@
+    scale: int,
+    create_using: Union[ar.Graph,ar.DiGraph,ar.PropGraph],
+    edge_factor: int = 16,
+    p: Union[float, Tuple[float]] = (0.57, 0.19, 0.19, 0.05),
+    weighted: bool = False,
+    permute: bool = True,
+    standardize: bool = True,
+) -> Union[ar.Graph,ar.DiGraph,ar.PropGraph]:
+    """
+    Recursive MATrix random graph generator.
+
+    Parameters
+    ----------
+    scale : int
+        number of nodes = 2 ** scale
+    edge_factor : int
+        each node has this many edges
+    p : { float | Tuple[float] }
+        link-formation probabilites. Single float interpreted as upper-left
+        quadrant probability with other quadrants equally sharing the
+        complement. Tuples will be intepreted as (a, b, c, d) as described in
+        the reference. Defaults to specification from Graph500.
+    weighted : bool (default False)
+        output uniformly random weights in [0, 1] for each edge.
+    permute : bool (default True)
+        randomly relabel nodes and permute edges
+    standardize : bool (default True)
+        standardize edges afterwards
+
+    create_using : Union[ar.Graph,ar.DiGraph,ar.PropGraph]
+        Arachne graph constructor
+        constructors supported
+        ar.Graph
+        ar.DiGraph
+        ar.PropGraph
+
+    Returns
+    -------
+    graph: Union[ar.Graph,ar.DiGraph,ar.PropGraph]
+
+    References
+    ----------
+    R-MAT: A Recursive Model for Graph Mining.
+        Deepayan Chakrabarti, Yiping Zhan and Christos Faloutsos.
+        Proceedings of the Fourth SIAM International Conference on Data Mining.
+        Apr 22-24 2004
+
+    Notes
+    -----
+    Stolen brazenly from arkouda/toys/connected_components.py and the Graph500
+    benchmark description.
+    """
+    n = 2 ** scale              # number nodes
+    m = n * edge_factor         # number edges
+
+    if isinstance(p, float) and 0 <= p <= 1:
+        a = p
+        b = c = d = (1.0 - p) / 3.0
+    elif isinstance(p, tuple) and all(0 <= x <= 1 for x in p) and sum(p) == 1:
+        a, b, c, d = p
+    else:
+        raise ValueError(f"p = {p} doesn't represent valid probability for RMAT.")
+    ab, cNorm, aNorm = a + b, c / (c + d), a / (a + b)
+
+    V, U = ak.zeros(m, dtype='int64'), ak.zeros(m, dtype='int64')
+    for i in range(scale):
+        vMask = ak.randint(0, 1, m, dtype='float64') > ab
+        uMask = (ak.randint(0, 1, m, dtype='float64')
+                 > (cNorm * vMask + aNorm * (~vMask)))
+        V += vMask * (2 ** i)
+        U += uMask * (2 ** i)
+
+    if permute:
+        # permute vertex labels
+        pi = get_perm(n)
+        V, U = pi[V], pi[U]
+
+        # permute edges
+        pi = get_perm(m)
+        V, U = V[pi], U[pi]
+
+    if weighted:
+        W = ak.uniform(V.size)
+        graph = empty_graph(create_using)
+        if standardize:
+            graph.add_edges_from(V, U, W)
+        return graph
+    else:
+        graph = empty_graph(create_using)
+        if standardize:
```

Comment:
You don't need to check standardize here, or include it as a parameter. Since graphs and directed graphs in Arachne don't store multiple edges then standardization will always happen so we should always use `add_edges_from`. Also, since these methods all return graphs, then standardization will always take place.